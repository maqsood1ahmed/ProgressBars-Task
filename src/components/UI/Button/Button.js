import React from "react";

import "./Button.scss";
import propTypes from "prop-types";

const Button = (props) => {
    return (<button value={props.value} onClick={(value) => props.handleClick(value)} className="button-progressbar">{props.text}</button>);
};

Button.propTypes = {
    text: propTypes.string.isRequired,
    value: propTypes.number.isRequired,
    handleClick: propTypes.func
};

export default Button;
