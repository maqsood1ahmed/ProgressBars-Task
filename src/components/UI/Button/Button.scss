.button-progressbar{
    display: flex;
    padding:10px 20px;
	display:inline-block;
	border-radius:3px;
	background:#ccc;
	font-size:16px;
    color:#333;
    padding:6px 30px;
	font-size:22px;
}