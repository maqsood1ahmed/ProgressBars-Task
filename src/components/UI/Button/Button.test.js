import React from 'react';

import { configure, shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';

import Button from "./Button";

configure({ adapter: new Adapter() });

describe('Button test', () => {
    it('should be defined', () => {
        expect(Button).toBeDefined();
    });
    it('should render correctly', () => {
        const tree = shallow(
            <Button text="-10" value={-10}/>
        );
        expect(tree).toMatchSnapshot();
    });
});
