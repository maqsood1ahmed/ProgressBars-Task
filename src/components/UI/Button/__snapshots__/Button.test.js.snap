// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button test should render correctly 1`] = `ShallowWrapper {}`;
